PSSwagger/PSSwagger.psm1

```diff
@@ -265,39 +271,54 @@ function New-PSSwaggerModule
         Write-Verbose -Message $message
         
         $ev = $null
-        Invoke-WebRequest -Uri $SpecificationUri -OutFile $SpecificationPath -ErrorVariable ev
-        if($ev) {
+
+        $webRequestParams = @{
+            'Uri' = $SpecificationUri
+            'OutFile' = $SpecificationPath
+        }
+
+        if($Credential -ne $null) {
+            $webRequestParams['Credential'] = $Credential
+        }
+
+        Invoke-WebRequest @webRequestParams -ErrorVariable ev 
+        if ($ev) {
             return 
         }
 
         $jsonObject = ConvertFrom-Json -InputObject ((Get-Content -Path $SpecificationPath) -join [Environment]::NewLine) -ErrorAction Stop
-        if((Get-Member -InputObject $jsonObject -Name 'Documents') -and ($jsonObject.Documents.Count))
-        {
+        if ((Get-Member -InputObject $jsonObject -Name 'Documents') -and ($jsonObject.Documents.Count)) {
             $AutoRestModeler = 'CompositeSwagger'
             $BaseSwaggerUri = "$SpecificationUri".Substring(0, "$SpecificationUri".LastIndexOf('/'))
-            foreach($document in $jsonObject.Documents)
-            {
+            foreach ($document in $jsonObject.Documents) {
                 $FileName = Split-Path -Path $document -Leaf
                 $DocumentFolderPrefix = (Split-Path -Path $document -Parent).Replace('/', [System.IO.Path]::DirectorySeparatorChar).TrimStart('.')
                 
                 $DocumentFolderPath = Join-Path -Path $TempPath -ChildPath $DocumentFolderPrefix
 
-                if(-not (Test-Path -LiteralPath $DocumentFolderPath -PathType Container))
-                {
+                if (-not (Test-Path -LiteralPath $DocumentFolderPath -PathType Container)) {
                     $null = New-Item -Path $DocumentFolderPath -ItemType Container -Force -Confirm:$false -WhatIf:$false
                 }
                 $SwaggerDocumentPath = Join-Path -Path $DocumentFolderPath -ChildPath $FileName
 
                 $ev = $null
-                Invoke-WebRequest -Uri $($BaseSwaggerUri + $($document.replace('\','/').TrimStart('.'))) -OutFile $SwaggerDocumentPath -ErrorVariable ev
-                if($ev) {
+                $webRequestParams = @{
+                    'Uri' = $($BaseSwaggerUri + $($document.replace('\', '/').TrimStart('.')))
+                    'OutFile' = $SwaggerDocumentPath
+                }
+        
+                if($Credential -ne $null) {
+                    $webRequestParams['Credential'] = $Credential
+                }
```

Comment:
This is not needed as Credential is already assigned prior to this foreach loop.

Reply:
I have a fix coming in that will just re-use $webRequestParams